### PR TITLE
Fix range in clearExtraData

### PR DESCRIPTION
**Pull request by JGEthanChen**
It would go wrong if the `oldEndKey` is "", and `newEndKey` is not "". 
For example, if the `oldEndKey` is "", and `newEndKey` is "3". After comparing, clear would not happen because "3" > "".  Perhaps add one condition like this would help.

### CODE REVIEW

File: kv/raftstore/peer_storage.go [hunk 1]
```diff
@@ -261,7 +261,7 @@ func (ps *PeerStorage) clearExtraData(newRegion *metapb.Region) {
 	if bytes.Compare(oldStartKey, newStartKey) < 0 {
 		ps.clearRange(newRegion.Id, oldStartKey, newStartKey)
 	}
-	if bytes.Compare(newEndKey, oldEndKey) < 0 {
+	if bytes.Compare(newEndKey, oldEndKey) < 0 || len(oldEndKey) == 0 {
```
Comment (Connor1996) [inline kv/raftstore/peer_storage.go]: what if both `newEndKey` and `oldEndKey` are empty?

Comment (JGEthanChen) [reply kv/raftstore/peer_storage.go]: If two both are empty, it shouldn't call `clearRange()`.  
Maybe change like this?
```suggestion
	if bytes.Compare(newEndKey, oldEndKey) < 0 || (len(oldEndKey) == 0 && len(newEndKey) != 0) {
```

Comment (Connor1996) [reply kv/raftstore/peer_storage.go]: looks good

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by tisonkun — COMMENTED** [review]
@Whorra can you add a test to verify the changes?

**Pull request merged**